### ISSUE TITLE
Support specifying AMI family release version in provisioner

### ISSUE THREAD

**Issue by stevehipwell** [aws/karpenter-provider-aws#1495]
**Tell us about your request**
When specifying an AMI family in the provisioner I'd also like to be able to specify the release version, with the default behaviour falling back to the latest available version. This might need to be modelled as a map if the provisioner is supporting multiple families so each ones version can be specified.

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
We need to validate each AMI release before we're willing to allow any of or clusters to use it, we've been caught out too many times by AMIs that cause issues. We also want to be able to keep our node group node versions aligned with our Karpenter versions.

**Are you currently working around this issue?**
n/a

**Additional context**
This is related to #1008 and the logic for this could be used when an AMI ID is specified which is an official one.

**Attachments**
n/a

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by njtran** [comment]
Sorry for the late response on this. We're thinking of implementing this as part of an extension of the current AMI controls after implementing a mechanism for enforcing AMI Drift [here](https://github.com/aws/karpenter/issues/1716). Linking it in the #1738 issue as well for tracking.

**Comment by njtran** [comment]
Hey all, if you're tracking this, come give some feedback on https://github.com/aws/karpenter-core/pull/207!

**Comment by stevehipwell** [comment]
@njtran I think this has been implemented hasn't it?

**Comment by jonathan-innis** [comment]
> I think this has been implemented hasn't it

This isn't in there yet. We pin to the K8s control-plane minor version but still auto-upgrade to the latest patch without giving any control just based on a semver selector. You can really only pin today by grabbing the AMI id and pinning that in your `amiSelectorTerms` or you are just forced to take the latest patch version.

If you are using your own AMIs, you're in a bit of a worse spot since you would have to tag the version using image tags on the image that you created AND you would have to specify that same version tag in the `amiSelectorTerms`. There are some ideas here around templating the discovered control-plane K8s version so you could do something like the example below.

```
amiSelectorTerms:
  - version: "{{ .ClusterVersion.Major}}.{{ .ClusterVersion.Minor }}"
```

Basically, allowing the user themselves to define the tag key and then template out the discovered bits within karpenter's runtime itself.

**Comment by njtran** [comment]
Just adding a related issue on here: https://github.com/aws/karpenter/issues/4769. 

I could see the version bit for kubernetes versions, but there's also a question on how we'd be able to pin a EKS Optimized AMI Version. I think this would be easier to reason about if all EKS Optimized AMIs used semver, but I think only Bottlerocket does.

**Comment by njtran** [comment]
This is completed and will be included in v1. https://github.com/aws/karpenter-provider-aws/pull/6450
